massa-network-worker/src/network_worker.rs

```diff
@@ -153,62 +230,69 @@ impl NetworkWorker {
                         }
                     });
                 }
-                need_connect_retry = false;
+
+                // Spawn an async task to handling out connection futures.
+                let out_connection_tx_clone = out_connection_tx.clone();
+                out_connection_handle = Some(self.runtime.spawn(async move {
+                    while let Some((ip_addr, res)) = out_connecting_futures.next().await {
+                        let out_connection_tx_clone = out_connection_tx_clone.clone();
+                        Handle::current()
```

Comment:
Maybe this comment worth nothing but I find it very hard to read and understand the fact that the channel aren't async but they are used in the async context. The syntax also seems complicated to me.

Reply:
> I find it very hard to read and understand the fact that the channel aren't async but they are used in the async context

I suggest you read the docs at https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html

Reply:
I see. I think I understand it just "syntaxly speaking" and visualizing it is pretty hard to me.